### PR TITLE
Add ruff source to none-ls

### PR DESCRIPTION
**Pull request by nfelber**
Thank you for the amazing project :)

This adds the ruff linter and formatter as a source for none-ls.
I basically copied similar previous PRs and tested locally, so please let me know if you see any issue.

### CODE REVIEW

File: plugins/none-ls/servers.nix [hunk 1]
```diff
@@ -93,6 +93,9 @@ with lib; let
       staticcheck = {
         pacakge = pkgs.go-tools;
       };
+      ruff = {
+        package = pkgs.ruff;
+      };
```
Comment (GaetanLepage) [inline plugins/none-ls/servers.nix:98]:
```suggestion
      ruff = {
        package = pkgs.ruff;
      };
```
Please, preserve ordering (`staticcheck` should go below `ruff`)

File: tests/test-sources/plugins/none-ls.nix [hunk 2]
```diff
@@ -71,6 +71,7 @@
           ltrs.enable = true;
           markdownlint.enable = true;
           shellcheck.enable = true;
+          ruff.enable = true;
```
Comment (GaetanLepage) [inline tests/test-sources/plugins/none-ls.nix:74]:
```suggestion
          ruff.enable = true;
          shellcheck.enable = true;
```

Same here

File: plugins/none-ls/servers.nix (hunk 1 above)
Comment (nfelber) [reply plugins/none-ls/servers.nix]: Sorry I missed that! Fixed

**Review by GaetanLepage — APPROVED** [review]
All good !
Thanks for your contribution :)

**Pull request merged**